src/main/java/duke/main/Duke.java

```diff
@@ -0,0 +1,166 @@
+import duke.task.Task;
+import duke.task.Todo;
+
+
+
+public class Duke {
+    private final Storage storage;
+    private final Ui ui;
+    private TaskList tasks;
+
+    /**
+     * Create duke object.
+     * @param filePath Specify storage file path.
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = storage.load();
+        } catch (FileNotFoundException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Adds a new task depending on type and append in to tasks.
+     *
+     * @param str       the whole input string, split.
+     * @throws TaskException if no details when adding.
+     */
+    private void addTask(String[] str) throws TaskException {
+        if (str[0].compareTo("deadline") == 0) {
+            if (str.length == 1) {
+                throw new TaskException("deadline");
+            }
+
+            StringBuilder stuff = new StringBuilder();
+            StringBuilder deadline = new StringBuilder();
+
+            for (int i = 1; i < str.length; i++) {
+                if (str[i].compareTo("/at") == 0) {
+                    for (int j = 1; j < i; j++) {
+                        stuff.append(str[j]).append(" ");
+
+                    }
+                    for (int j = i + 1; j < str.length; j++) {
+                        deadline.append(str[j]).append(" ");
+                    }
+                    break;
+                }
+            }
+            tasks.add(new Deadline(stuff.toString(), Parser.parseDate(deadline.toString())));
+        } else if (str[0].compareTo("todo") == 0) {
+            if (str.length == 1) {
+                throw new TaskException("todo");
+            }
+
+            StringBuilder stuff = new StringBuilder();
+            for (int i = 1; i < str.length; i++) {
+                stuff.append(str[i]).append(" ");
+            }
+            tasks.add(new Todo(stuff.toString()));
+        } else if (str[0].compareTo("event") == 0) {
+            if (str.length == 1) {
+                throw new TaskException("event");
+            }
+
+            StringBuilder stuff = new StringBuilder();
+            StringBuilder deadline = new StringBuilder();
+
+            for (int i = 1; i < str.length; i++) {
+                if (str[i].compareTo("/at") == 0) {
+                    for (int j = 1; j < i; j++) {
+                        stuff.append(str[j]).append(" ");
+                    }
+                    for (int j = i + 1; j < str.length; j++) {
+                        deadline.append(str[j]).append(" ");
+                    }
+                    break;
+                }
+            }
+            tasks.add(new Event(stuff.toString(), Parser.parseDate(deadline.toString())));
+        } else {
+            throw new TaskException("tasks");
+        }
+
+        System.out.println("Got it. I've added this duke.task: ");
```

Comment:
There are a lot of println in the Duke class, perhaps output can be handled by the UI class?